File: 1002/linklist.py
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None


def create():
    head = None
    pre = None

    # 读入第一个节点数据
    data = int(input())

    while data != 0:
        cur = Node(data)  # 创建新节点

        if head is None:
            head = cur  # 第一个节点作为头节点
        else:
            pre.next = cur  # 链接前驱节点与当前节点

        pre = cur  # 更新前驱节点
        data = int(input())  # 继续输入数据

    return head


def print_list(n):
    while n:
        print(n.data, end=" ")  # 打印当前节点数据
        n = n.next  # 移动到下一个节点
    print()


# 主函数
if __name__ == "__main__":
    head = create()  # 创建链表
    print_list(head)  # 打印链表

File: 1001/LinkList.cpp
#include <iostream>
using namespace std;

struct node
{
	int data;
	node* next;
};


node* Create()
{
	node *cur, *pre;
	node* head = NULL;
	cur = new node;
	cin>>cur->data;
	while (cur->data != 0)
	{
		if (head == NULL)
		{
			head = cur;
		}
		else 
		{
			pre->next = cur;
		}
		pre = cur;
		cur = new node;
		cin>>cur->data;
	}
	pre->next = NULL;
	delete cur;
	return head;
}

void insert(int n, int c, node *temp)
{
	// 把元素内容放进节点里面
	node *new_node = new node;
	new_node->data = c;
	// 移动temp到n处
	for (int i = 0; i < n; i++) {
		temp = temp->next;
	}
	// 插入
	new_node->next = temp->next;
	temp->next = new_node;
}

void del(int n, node *head) {
	node *p = head;
	for (int i = 0; i < n-1; i++) {
		head = head->next;
		p = p->next;
	}
	head = head->next;
	p->next = head->next;
	delete head;
}

void print(node* n)
{
	while(n)
	{
		printf("%d", n->data);
		n = n->next;
	}
}


int main()
{
	node* head;
	head = Create();
	insert(2, 4, head);
	del(4, head);
	print(head);
	return 0;
}
